libs/wingc/src/diagnostic.rs

```diff
@@ -285,10 +276,17 @@ pub fn report_diagnostic(diagnostic: Diagnostic) {
 	// If we're running in wasm32 then send the diagnostic to the client
 	#[cfg(target_arch = "wasm32")]
 	{
-		let json = serde_json::to_string(&diagnostic).unwrap();
-		let bytes = json.as_bytes();
-		unsafe {
-			send_diagnostic(bytes.as_ptr(), bytes.len() as u32);
+		match serde_json::to_string(&diagnostic) {
+			Ok(json) => {
+				let bytes = json.as_bytes();
+				unsafe {
+					send_diagnostic(bytes.as_ptr(), bytes.len() as u32);
+				}
+			}
+			Err(err) => {
+				eprintln!("Error serializing diagnostic: {}", err);
+				// avoiding a panic here because we don't want to crash the panic handler
+			}
```

Comment:
ℹ️  When having issue with memory corruption, this section would panic due to the json being invalid. This shouldn't normally happen, but just in case.